mantidimaging/gui/windows/live_viewer/model.py

```diff
@@ -29,85 +93,125 @@ def path(self):
         """
         Find all the images in the directory and emit the
         image_changed signal for the last modified image.
         """
         self.images = self._get_image_files()
 
-    def find_last_modified_image(self):
+    def sort_images_by_modified_time(self, images: list[Image_Data]) -> list[Image_Data]:
         """
-        Find the last modified image in the directory and
-        emit the image_changed signal.
+        Sort the images by modified time.
+
+        :param images: list of image objects to sort by modified time
+        :return: sorted list of images
         """
+        return sorted(images, key=lambda x: x.image_modified_time)
 
+    def get_images(self):
+        """Return the sorted images"""
         if self.images:
             try:
-                last_modified_image = max(self.images, key=lambda x: x.stat().st_mtime)
+                return self.images
             except FileNotFoundError:
-                last_modified_image = ''
-
-            self.image_changed.emit(str(last_modified_image))
-
-    def _handle_directory_change(self, directory):
-        self.find_images()
-        if self.images and self._validate_images(self.images):
-            self.find_last_modified_image()
-        else:
-            self.image_changed.emit('')
+                return []
 
-    def _validate_file(self, file_path) -> bool:
+    def _handle_directory_change(self, directory) -> None:
         """
-        Check if a file is valid.
-        """
-        return file_path.is_file() and self._is_image_file(file_path.name)
+        Handle a directory change event. Update the list of images
+        to reflect directory changes and emit the image_changed signal
+        for the last modified image.
 
-    def _validate_images(self, images: list) -> bool:
-        """
-        Check if a list of images are valid.
+        :param directory: directory that has changed
         """
-        return all(self._validate_file(image) for image in images)
+        try:
+            self.find_images()
+            self.image_changed.emit(self.images)
+        except FileNotFoundError:
+            self.image_changed.emit([])
 
     def _get_image_files(self):
-        image_files = []
-        for file_path in Path(self.directory).iterdir():
-            if not self._validate_file(file_path):
-                LOG.debug(f'INVALID FILE: {file_path} is not valid an image file')
-                continue
-            file_size = file_path.stat().st_size
-            if file_size > 45 and self._validate_file(file_path):
-                image_files.append(file_path)
-            else:
-                LOG.debug(f'INVALID FILE: {file_path} is not valid an image file or is empty')
-        return image_files
+        image_files = [
+            image_obj for file_path in Path(self.directory).iterdir()
+            if self._is_image_file(file_path.name) and (image_obj := Image_Data(file_path)).image_size > 45
+        ]
```

Comment:
I think the try/except need to be inside the loop. Otherwise one bad file causing an exception in `Image_Data(file_path)` will jump us out of ` _get_image_files()` so will result in no images being found.